homeassistant/helpers/debounce.py

```diff
@@ -76,3 +96,11 @@ def async_cancel(self) -> None:
             self._timer_task = None
 
         self._execute_at_end_of_timer = False
+
+    @callback
+    def _schedule_timer(self) -> None:
+        """Schedule a timer."""
+        self._timer_task = self.hass.loop.call_later(
+            self.cooldown,
+            lambda: self.hass.async_create_task(self._handle_timer_finish()),
+        )
```

Comment:
Hi @balloob,

As `self.hass.loop.call_later` does not register in `self._pending_tasks`, the calls in tests to `self.hass.block_till_done()` do not wait for these timers, so deboucer with immediate=False fail in tests :(

Something like the dumb `self.hass.async_create_task(self._wait_until_cooldown())` from #32155 is needed, right?

Reply:
Actually `async_block_till_done` takes care of that, hence why my tests pass :)

https://github.com/home-assistant/home-assistant/blob/ba4cc373c8696cc25b7e6084c6026944eb7299c0/homeassistant/core.py#L382-L385

Reply:
That does the trick for any pending 'call_soon_threadsafe', but the _call_later_ is not going to start running with that, right?

I had to introduce the _wait_until_cooldown_ in the other PR **to fix the statistics tests**: without it, all of them failed. 

(if switching to `immediate=True` this is not an issue, as the call to update is made on time, but for the stats I decided to use the delayed call).

Reply:
oh you're right. I mimic that in the test by calling the timer done function.